### PR TITLE
feat(UST-1264): create `@vue-storefront/changesets` package

### PR DESCRIPTION
**Pull request by jagoral**
### 🔗 Linked issue

- [UST-1264](https://vsf.atlassian.net/browse/UST-1264)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Created `@vue-storefront/changesets` that provides a custom changelog format for the `changesets` package.
- Unfortunately the changeset’s has a [very limited customization options](https://github.com/changesets/changesets/issues/1178), so we’ll still have to take care about most of the the things manually.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

[UST-1264]: https://vsf.atlassian.net/browse/UST-1264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/changesets/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+```sh
+yarn add -D @vue-storefront/changesets
+```
+
+## Usage
+
+Add the following to your `changeset/config.json` file:
+
+```json[.changeset/config.json]
```
Comment (FRSgit) [inline packages/changesets/README.md:17]: Question: out of curiosity - is that info about filepath visible anywhere? I don't see it in the github preview, but maybe I'm missing something?

File: packages/changesets/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+In the changeset's release PR makes sure, that the migration guide is provided in the CHANGELOG.md file.
+
+## Conventions
+
+- [ADDED] { what was added, with link to the docs }, { what new capabilities it introduced }{ example usage }
+- [CHANGED] { what has changed }, { why it has changed}{ before / after code }
+- [FIXED] { what was fixed }, { what was the previous behaviour}. { what is the new behaviour }
+- [REMOVED] { what was removed }, { why }. { additional context for ones using this feature eg. it is now handled in a different way in a different place }
```
Comment (FRSgit) [inline packages/changesets/README.md:53]: e.g. dependency upgrade with breaking changes should fall under "changed"?

File: packages/changesets/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { changelogFunctions } from "./changelog";
+
+export default changelogFunctions;
```
Comment (FRSgit) [inline packages/changesets/src/index.ts:3]: Nitpick: I think you could also write it as 
```suggestion
export { changelogFunctionsa as default } from "./changelog";
```

File: packages/changesets/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "@vue-storefront/integrations-tsconfig/tsconfig",
+  "compilerOptions": {
```
Comment (FRSgit) [inline packages/changesets/tsconfig.json:3]: Nitpick: Please? 🙏 😄 
```suggestion
{
  "extends": "@vue-storefront/integrations-tsconfig/tsconfig",
  "compilerOptions": {
    "strict": true,
```

File: packages/changesets/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+## Usage
+
+Add the following to your `changeset/config.json` file:
+
+```json[.changeset/config.json]
+{
+  "changelog": "@vue-storefront/changesets",
+}
```
Comment (FRSgit) [inline packages/changesets/README.md:20]: Noice, cannot be easier 👍

File: packages/changesets/tsconfig.json (hunk 4 above)
Comment (jagoral) [reply packages/changesets/tsconfig.json:3]: sure! :D

File: packages/changesets/README.md (hunk 1 above)
Comment (jagoral) [reply packages/changesets/README.md]: in our docs yes (I think `@nuxt/content` adds it as a wrapper for shiki), but I see that gh doesn't have this feature :(

File: packages/changesets/README.md (hunk 2 above)
Comment (jagoral) [reply packages/changesets/README.md:53]: for the deps update, changesets creates the changelog automatically 
https://github.com/vuestorefront/vue-storefront/blob/851e7b0dcbc85013592edbddd247e456e6d1e7b6/packages/changesets/src/__tests__/changelog.spec.ts#L67

File: packages/changesets/src/index.ts (hunk 3 above)
Comment (jagoral) [reply packages/changesets/src/index.ts:3]: Initially I commited the suggestion, but then I saw that it conflicts with `no-restricted-exports` rule
https://eslint.org/docs/latest/rules/no-restricted-exports

**Review by WojtekTheWebDev — APPROVED** [review]

**Pull request merged**